### PR TITLE
Admin bar: move the Reader admin bar item to the secondary group

### LINKED ISSUE #93534: Reader - move admin bar link to the right side of the masterbar
related thread - p1723032282006269-slack-C06DN6QQVAQ

We want to test having the reader on the right side of the adminbar instead of the left.

Current position:
Screenshot 2024-08-14 at 11 51 19 AM

Proposed position:
image

We will want to ensure this is moved for:

- [ ] calypso admin bar
- [ ] site admin bar (classic)
- [ ] site admin bar (default)

We will want to also measure the impact of this on the readers usage. Notes on click events and feed loading events that may be useful in the slack thread p1723488532296959/1723032282.006269-slack-C06DN6QQVAQ

### PR DESCRIPTION
**Pull request by allilevine**
Fixes https://github.com/Automattic/wp-calypso/issues/93534

- Related Calypso PR: https://github.com/Automattic/wp-calypso/pull/93749

## Proposed changes:

* Move the Reader admin bar item to the right side of the admin bar (the secondary group).
* Give the Help Center item a priority.

Simple before:
Screen Shot 2024-08-21 at 4 45 12 PM

Simple this branch:
Screen Shot 2024-08-21 at 4 44 58 PM

Atomic before:
Screen Shot 2024-08-21 at 4 53 14 PM

Atomic this branch:
Screen Shot 2024-08-21 at 4 53 38 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/wp-calypso/issues/93534

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

* To test on WoA, go to the Plugins menu on a WordPress.com Atomic site. Click on the "Upload" button and follow the upgrade flow to be able to upload, install, and activate [the Jetpack Beta plugin](https://jetpack.com/download-jetpack-beta/). Once the plugin is active, go to Jetpack > Jetpack Beta, select your plugin, and enable the update/admin-menu-reader-placement branch.
* Also add define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true ); to your wp-config.php file.
* Go to WP Admin.
* Verify that the Reader menu item is on the right, to the left of Help Center.
* Check that clicking the Reader item takes you to the Reader.
* Test on mobile screen sizes.
* Test on supported browsers.
* Test with different color schemes.
* Test with a RTL language.
* Repeat the testing steps with a Simple site. To test on Simple, run the following command on your sandbox:

bin/jetpack-downloader test jetpack-mu-wpcom-plugin update/admin-menu-reader-placement

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php [hunk 1]
```diff
@@ -100,7 +100,8 @@ function ( $wp_admin_bar ) {
 							),
 						)
 					);
-				}
+				},
+				119
```
Comment (allilevine) [inline projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php]: Order the help center after the Reader, but before Notifications.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php [hunk 2]
```diff
@@ -144,18 +144,22 @@ function wpcom_replace_wp_logo_with_wpcom_all_sites_menu( $wp_admin_bar ) {
...
+			'id'     => 'reader',
+			'title'  => '<span class="ab-icon" aria-hidden="true"></span><span class="screen-reader-text">' .
+						/* translators: Hidden accessibility text. */
+						__( 'Reader', 'jetpack-mu-wpcom' ) .
+						'</span>',
+			'href'   => maybe_add_origin_site_id_to_url( 'https://wordpress.com/read' ),
+			'meta'   => array(
 				'class' => 'wp-admin-bar-reader',
 			),
+			'parent' => 'top-secondary',
```
Comment (allilevine) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php:158]: Add the Reader to the secondary admin bar group.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php [hunk 3]
```diff
@@ -144,18 +144,22 @@ function wpcom_replace_wp_logo_with_wpcom_all_sites_menu( $wp_admin_bar ) {
...
 function wpcom_add_reader_menu( $wp_admin_bar ) {
-	$wp_admin_bar->add_node(
+	$wp_admin_bar->add_menu(
 		array(
-			'id'    => 'reader',
-			'title' => __( 'Reader', 'jetpack-mu-wpcom' ),
-			'href'  => maybe_add_origin_site_id_to_url( 'https://wordpress.com/read' ),
-			'meta'  => array(
+			'id'     => 'reader',
+			'title'  => '<span class="ab-icon" aria-hidden="true"></span><span class="screen-reader-text">' .
```
Comment (allilevine) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php:150]: I'm following the example of the W logo here.

**Review by fushar — COMMENTED** [review]
Can we fix the gap between Reader and Help Center icon? On >480px mobile width, it seems to have narrower gap than the others:

image

**Comment by allilevine** [comment]
> Can we fix the gap between Reader and Help Center icon? On >480px mobile width, it seems to have narrower gap than the others:
> 
> image

Thanks, I'll take a look!

**Comment by allilevine** [comment]
@fushar I pushed a change, let me know how it looks. I changed both < 782px and < 480px styles to get it to match the Help Center - Notifications gap.

Screen Shot 2024-08-21 at 2 47 06 PM

**Review by fushar — DISMISSED** [review]
I pushed a small CSS fix.

It's now pixel perfect with Calypso on various screen widths! Tested on Atomic and Simple as well.

**Comment by allilevine** [comment]
> I pushed a small CSS fix.
> 
> It's now pixel perfect with Calypso on various screen widths! Tested on Atomic and Simple as well.

Looks great, thanks @fushar!

File: projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php [hunk 4]
```diff
@@ -100,7 +100,8 @@ function ( $wp_admin_bar ) {
 							),
 						)
 					);
-				}
+				},
+				12
```
Comment (Addison-Stavlo) [inline projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php:105]: Is this priority change to ensure its left/right position wrt to the new help item?

**Review by Addison-Stavlo — COMMENTED** [review]
This looks good to me!

Non blocking - there are minor differences in styles compared to the calypso PR. Note on mobile that there is an extra padding division between the reader and the help center in wp-admin but not calypso. And the calypso one is a little wider. But these are very minor things I wouldn't expect folks to notice.

minor-differences

**Review by Addison-Stavlo — DISMISSED** [review]
WFM! Tested simple and atomic.

File: projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php (hunk 4 above)
Comment (allilevine) [reply projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php:105]: @Addison-Stavlo Yes! I guess [my earlier comment](https://github.com/Automattic/jetpack/pull/38976#discussion_r1723884966) disappeared. Previously it was the default priority, 10. To make sure they're next to each other, I gave Reader an 11 priority and Help Center a 12 priority. Let me know if you have another recommendation!

**Comment by allilevine** [comment]
> Non blocking - there are minor differences in styles compared to the calypso PR. Note on mobile that there is an extra padding division between the reader and the help center in wp-admin but not calypso. And the calypso one is a little wider. But these are very minor things I wouldn't expect folks to notice.

I'll timebox and see if there's a fix.

**Comment by allilevine** [comment]
@Addison-Stavlo I moved the padding to the inner span and I'm now seeing the same width as Calypso, no gap:


https://github.com/user-attachments/assets/b2604d05-334d-46a2-9d5d-83a3779dca28

**Review by fushar — DISMISSED** [review]
I pushed a small fix. It's now consistent with the corresponding Calypso PR 👍 

### 783px

|||
|-|-|
|wp-admin|image|
|Calypso|image|
||image|


### 600px

|||
|-|-|
|wp-admin|image|
|Calypso|image|
||image|

### 400px

|||
|-|-|
|wp-admin|image|
|Calypso|image|
||image|

### 350px

|||
|-|-|
|wp-admin|image|
|Calypso|image|

**Review by arthur791004 — DISMISSED** [review]
Nice 👍

**Review by fushar — DISMISSED** [review]

File: projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php (hunk 4 above)
Comment (Addison-Stavlo) [reply projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php:105]: Not off the top of my head, that makes sense to me!

Perhaps a comment to help context for the next code reader would be helpful. If we want the priorities to stay close to each other, we may want to add a small comment around both of them.

Comment (allilevine) [reply projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php:105]: Added, thanks!

**Review by Addison-Stavlo — APPROVED** [review]

**Pull request merged**